src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java

```diff
@@ -130,18 +171,19 @@ private HandleResult convertRecursively(ExecutionResultNode node,
                                             boolean onlyChildren,
                                             NadelContext nadelContext,
                                             TransformationMetadata transformationMetadata,
-                                            AtomicInteger nodeCount) {
-        HandleResult handleResult = convertSingleNode(node, parentNode, executionId, root, normalizedRootField, overallSchema, isHydrationTransformation, batched, fieldIdToTransformation, typeRenameMappings, onlyChildren, nadelContext, transformationMetadata, nodeCount);
+                                            AtomicInteger nodeCount, AtomicInteger fieldRenameCount, AtomicInteger typeRenameCount) {
```

Comment:
See here about an encapsulating class makes it nicer

Reply:
done